File: simpleMobileVMApp/simpleMobileVMAndroidLib/src/main/java/com/slickpath/mobile/android/simple/vm/DefaultInputListener.java
package com.slickpath.mobile.android.simple.vm;

import java.io.IOException;

/**
 * DefaultInputListener that uses the console input via System.in.read
 *
 * @author Pete Procopio
 */
public class DefaultInputListener implements InputListener {

    @Override
    public int getInt() throws IOException {
        return System.in.read();
    }

    @Override
    public char getChar() throws IOException {
        return (char) System.in.read();
    }
}

File: simpleMobileVMApp/simpleMobileVMAndroidLib/src/main/java/com/slickpath/mobile/android/simple/vm/DefaultInputListener.kt
package com.slickpath.mobile.android.simple.vm

import java.io.IOException

/**
 * DefaultInputListener that uses the console input via System.in.read
 *
 * @author Pete Procopio
 */
class DefaultInputListener : InputListener {

    @get:Throws(IOException::class)
    override val int: Int
        get() = System.`in`.read()

    @get:Throws(IOException::class)
    override val char: Char
        get() = System.`in`.read().toChar()
}